mobilecoind-json/src/data_types.rs

```diff
@@ -486,6 +486,39 @@ impl TryFrom<&JsonOutlay> for api::Outlay {
     }
 }
 
+#[derive(Deserialize, Serialize, Default, Debug, Clone)]
+pub struct JsonOutlayV2 {
+    pub value: JsonU64,
+    pub receiver: JsonPublicAddress,
+    pub token_id: JsonU64,
+}
```

Comment:
It is used in all of the historical request objects, see `src/bin/main.rs`

Reply:
I think it's more error prone if we don't mirror the proto types into json separately. There's also issues around, if we have conversions from `JsonOutlay` to both `Outlay` and `OutlayV2` it can be a lossy conversion and hide a bug that causes people to send the wrong token.

We do have to have Outlay and OutlayV2 in proto or I'm think we will break backwards compat.

Reply:
Shouldn't the proto definition be extendable? Normally optional fields can be added I think. And then we can have the default behavior when token_id isn't defined. Which is what we want.

Reply:
See my earlier comments on the PR description:

> Outlay is used both in the requests and responses for generate_transaction, so what I decided to do is extend Outlay to OutlayV2, which has a token id parameter, and TxProposal now contains OutlayV2. That protobuf extension was backwards compatible. All the historical requests, though, still use Outlay.

Extending outlay in the responses is fine -- extending outlay in the requests is not fine. The current semantics of all the requests in mobilecoind is that token id is a single top-level field that specifies the token id for everything in the entire transaction. If we change all of those to OutlayV2, we break old clients unless we ignore the new token id field in OutlayV2. That's confusing to the client and a hazard, so that's not an acceptable approach.

That means that we cannot completely get rid of the unextended version of Outlay. We replace it with OutlayV2 in responses, and use OutlayV2 in any new requests, but we don't migrate to the new Outlay in any pre-existing request objects, because we can't do so without breaking things.